espnet2/speechlm/core_lm/ar_parallel.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Jinchuan Tian
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+# Implementation of Parallel architecture: https://arxiv.org/pdf/2306.05284
+
+from typing import Dict, Tuple
+
+import torch
+
+from espnet2.speechlm.core_lm.abs_core_lm import AbsCoreLM
+from espnet2.speechlm.inference_utils import AbsInferenceConfig
+
+
+class ARParallelLM(AbsCoreLM):
```

Comment:
Is it used as an independent class, or is it only used for the ArDelayLM class?
If it is only used for the ArDelayLM class, we may merge this class with that.
If not, please explain it.